stock-managment/src/main/java/org/dhis2/android/rtsm/ui/managestock/components/ManageStockTable.kt

```diff
@@ -1,55 +1,48 @@
 package org.dhis2.android.rtsm.ui.managestock.components
 
-import android.content.Context
+import androidx.compose.material.BackdropScaffoldState
+import androidx.compose.material.ExperimentalMaterialApi
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.livedata.observeAsState
-import androidx.compose.ui.platform.LocalContext
 import com.google.android.material.composethemeadapter.MdcTheme
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
 import org.dhis2.android.rtsm.ui.managestock.ManageStockViewModel
-import org.dhis2.composetable.model.KeyboardInputType
-import org.dhis2.composetable.model.TableCell
-import org.dhis2.composetable.model.TextInputModel
 import org.dhis2.composetable.ui.DataSetTableScreen
 
+@OptIn(ExperimentalMaterialApi::class)
 @Composable
 fun ManageStockTable(
-    viewModel: ManageStockViewModel
+    viewModel: ManageStockViewModel,
+    backdropState: BackdropScaffoldState,
+    scope: CoroutineScope
 ) {
-    val context = LocalContext.current
     val screenState by viewModel.screenState.observeAsState()
 
     MdcTheme {
         DataSetTableScreen(
             tableScreenState = screenState!!,
             onCellClick = { _, cell ->
-                onCellClick(cell)
+                viewModel.onCellClick(cell)
             },
             onEdition = { isEditing ->
-                editingCellValue(context, isEditing)
+                editingCellValue(isEditing, backdropState, scope)
             },
-            onCellValueChange = { cell ->
-                onCellValueChanged(cell)
-            },
-            onSaveValue = { cell, _ ->
-                onSaveValueChange(cell)
-            }
+            onCellValueChange = viewModel::onCellValueChanged,
+            onSaveValue = viewModel::onSaveValueChange
         )
     }
 }
 
-fun editingCellValue(context: Context, editing: Boolean) {}
-
-fun onCellClick(cell: TableCell): TextInputModel {
-    return TextInputModel(
-        id = cell.id!!,
-        mainLabel = "Quantity",
-        currentValue = cell.value,
-        keyboardInputType = KeyboardInputType.NumericInput(
-            allowDecimal = false,
-            allowSigned = false
-        )
-    )
+@OptIn(ExperimentalMaterialApi::class)
+fun editingCellValue(
+    editing: Boolean,
+    backdropState: BackdropScaffoldState,
+    scope: CoroutineScope
+) {
```

Comment:
I think that you are coupling ManageStockTable with MainContent. You can pass here a lambda as paremeter like: 
```
fun editingCellValue(
    editing: Boolean,
    onEditionStart: () -> Unit
)
    if (editing) {
        onEditionStart.invoke()
    }
}
```
And the in MainContent: 
```
ManageStockTable(manageStockViewModel) {
                        scope.launch { backdropState.conceal() }
```

What do you think?